### PR TITLE
fcntl: Adding F_TRANSFEREXTENTS fcntl constant for macOs.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: src/fcntl.rs [hunk 1]
```diff
@@ -819,6 +819,12 @@ pub enum FcntlArg<'a> {
     /// is used as both IN/OUT as both its l2p_devoffset and
     /// l2p_contigbytes can be used for more specific queries.
     F_LOG2PHYS_EXT(&'a mut libc::log2phys),
+    /// Transfer any extra space in the file past the logical EOF
+    /// (as previously allocated via F_PREALLOCATE) to another file.
+    /// The other file is specified via a file descriptor as the lone extra argument.
+    /// Both descriptors must reference regular files in the same volume.
+    #[cfg(apple_targets)]
+    F_TRANSFEREXTENTS(RawFd),
```
Comment (SteveLauC) [inline src/fcntl.rs:827]: Hmm, just realized that I missed this type when added I/O safety to this module. 😵‍💫

**Review by SteveLauC — APPROVED** [review]
LGTM, let's wait for that libc PR, though Nix currently uses libc from branch `libc-0.2`, not `main`.

**Comment by devnexen** [comment]
Yes I planned to keep it as draft as long as it takes :)

**Comment by SteveLauC** [comment]
> Yes I planned to keep it as draft as long as it takes :)

That's totally fine:)

File: src/fcntl.rs [hunk 2]
```diff
@@ -952,6 +958,12 @@ pub fn fcntl<Fd: std::os::fd::AsFd>(fd: Fd, arg: FcntlArg) -> Result<c_int> {
             F_PREALLOCATE(st) => {
                 libc::fcntl(fd, libc::F_PREALLOCATE, st)
             },
+            #[cfg(apple_targets)]
+            F_TRANSFEREXTENTS(rawfd) => {
+                // FIXME: ongoing (PR)[https://github.com/rust-lang/libc/pull/3925/files]
```
Comment (SteveLauC) [inline src/fcntl.rs:963]: This PR has been merged and backported to the `libc-0.2` branch:)

File: src/fcntl.rs [hunk 3]
```diff
@@ -952,6 +958,12 @@ pub fn fcntl<Fd: std::os::fd::AsFd>(fd: Fd, arg: FcntlArg) -> Result<c_int> {
...
+                const F_TRANSFEREXTENTS: c_int = 110;
+                libc::fcntl(fd, F_TRANSFEREXTENTS, rawfd)
```
[the remark this answers is not in the document]
Comment (devnexen) [reply src/fcntl.rs:965]: Yes I forgot to change in fact..

File: src/fcntl.rs [hunk 4]
```diff
@@ -952,6 +958,11 @@ pub fn fcntl<Fd: std::os::fd::AsFd>(fd: Fd, arg: FcntlArg) -> Result<c_int> {
             F_PREALLOCATE(st) => {
                 libc::fcntl(fd, libc::F_PREALLOCATE, st)
             },
+            #[cfg(apple_targets)]
+            F_TRANSFEREXTENTS(rawfd) => {
+                // FIXME: ongoing (PR)[https://github.com/rust-lang/libc/pull/3925/files]
```
Comment (SteveLauC) [inline src/fcntl.rs:963]: Still has a comment to remove:)

**Review by SteveLauC — APPROVED** [review]
Thanks!

**Pull request merged**